### PR TITLE
Add skycli commands for multiple clusters

### PR DESCRIPTION
**Pull request by kiootic**
ref #400

### CODE REVIEW

**Comment by kiootic** [comment]
So, we'll have three commands?
- skycli config list-contexts
- skycli config use-context
- skycli config set-cluster

It seems a bit inconsistent to me, since set-cluster would manipulate contexts implicitly, and there is no set-context. That is why I choose to expose clusters in the commands only.

**Comment by louischan-oursky** [comment]
The problem is that what `skycli config set-cluster` is doing now is that it clears the config file and set up a context for the cluster.

skycli should always has a context to work with. A context is a combination of cluster and user. A context can without a user.

`skycli config set-cluster` adds a new cluster. If the cluster already exists, it is an error. Otherwise create a new context with the new cluster. (Should we switch to the newly created context?)

`skycli auth login` adds a new user. If the current context has no user, the context is updated in place. If the current context has user, create a new context with the cluster of the current context and the logged in user, switch to the new context.

skycli will have a global flag `--context` to override `current_context` in the config file.

Thoughts?

**Comment by kiootic** [comment]
The issue described that we'll support only one user per cluster for now, i.e. only one context per cluster. Is that correct?

Anyways, `set-cluster` need to be changed. I'm thinking of having `set-cluster` always update the context and cluster with specified name according to the input parameters, so it's an upsert operation to clusters and contexts with cluster name as key.

**Comment by kiootic** [comment]
I'll add the three commands mentioned before (`get-contexts`, `use-context`, set-cluster`). Contexts can be added/updated through `set-cluster`, and logging in would affect the user of current context.

**Comment by louischan-oursky** [comment]
Thanks. Can you also introduce the global `--context` flag? Actually it is original use case I want to support. It is particularly useful for `skycli app get-k8s-credentials`.

**Comment by louischan-oursky** [comment]
Maybe we should also review whether we still keep `skycli config view`

**Comment by kiootic** [comment]
> Maybe we should also review whether we still keep skycli config view

I would suggest making `skycli config view` output the config yaml verbatim.

File: features/270-next-skycli/commands.md [hunk 1]
```diff
@@ -66,13 +66,20 @@ View the current config.
...
-[app]
-app=myapp
+api_version: v1
+clusters:
+  - name: skygeario
+    cluster:
+      env: cloud
+      endpoint: 'https://controller.example.com'
+      api_key: api-key
+users: []
```
Comment (louischan-oursky) [inline features/270-next-skycli/commands.md]: For completeness, either include the user `skygeario` here or omit `user` in the context `skygeario`.

Comment (kiootic) [reply features/270-next-skycli/commands.md]: This output is possible, if there is no user associated with the `skygeario` context. In that case, the context would refer to an nonexistent user and `auth login` would create/update it.

File: features/270-next-skycli/commands.md [hunk 2]
```diff
@@ -64,26 +66,35 @@ View the current config.
...
+Configure Cluster endpoint and API key, it is provided by cluster admin.
+If there is no current context set, the configured Cluster would be set as the
+current context.
+If there exist a cluster with same name, it is updated with provided
+information.
 
 #### Flags
 
-- `--cluster=[CLUSTER_NAME]` Provide cluster name, currently only `skygeario` is supported.
+- `--cluster=[CLUSTER_NAME]` Provide cluster name.
```
Comment (carmenlau) [inline features/270-next-skycli/commands.md]: After this update, if user want to set their own cluster without prompt, they will need to provide `cluster` flag? If yes, please update the example below. Maybe `skycli config set-cluster --cluster=mycluster --endpoint=https://mycluster-controller --api-key=api_key` ?

**Pull request merged**